### PR TITLE
Endpoint to get all rates by source currency was added

### PR DESCRIPTION
**Pull request by DaniyarAskarov**
Issue: https://github.com/happy-travel/agent-app-project/issues/1577
Related PR: https://github.com/happy-travel/tsutsujigasaki-grpc-contracts/pull/5
Tested through debug and swagger

### CODE REVIEW

**Comment by DFSko** [comment]
need to update https://github.com/happy-travel/tsutsujigasaki-grpc-contracts too

File: HappyTravel.Tsutsujigasaki.Api/Controllers/RatesController.cs [hunk 1]
```diff
@@ -21,14 +22,32 @@ public RatesController(IRateService rateService)
...
 
 
+        /// <summary>
+        /// Returns conversion rates for the provided currency.
+        /// </summary>
+        /// <param name="sourceCurrency">The source currency code</param>
+        /// <returns></returns>
+        [ProducesResponseType(typeof(Dictionary<Currencies, decimal>), (int)HttpStatusCode.OK)]
+        [ProducesResponseType(typeof(ProblemDetails), (int)HttpStatusCode.BadRequest)]
+        [HttpGet("{sourceCurrency}")]
```
Comment (AlexeyShok) [inline HappyTravel.Tsutsujigasaki.Api/Controllers/RatesController.cs:32]: source-currency

File: HappyTravel.Tsutsujigasaki.Api/Controllers/RatesController.cs [hunk 2]
```diff
@@ -21,14 +22,32 @@ public RatesController(IRateService rateService)
...
+        public async Task<IActionResult> Convert([FromRoute] Currencies sourceCurrency)
```
Comment (AlexeyShok) [inline HappyTravel.Tsutsujigasaki.Api/Controllers/RatesController.cs:33]: Perhaps we also need to send a list of currencies to which we need a conversion rate?

File: HappyTravel.Tsutsujigasaki.Api/Controllers/RatesController.cs [hunk 3]
```diff
@@ -21,14 +22,32 @@ public RatesController(IRateService rateService)
...
+        {
+            var (_, isFailure, value, error) = await _rateService.Get(sourceCurrency);
+            if (isFailure)
+                return BadRequest(error);
+
+            return Ok(value);
```
Comment (AlexeyShok) [inline HappyTravel.Tsutsujigasaki.Api/Controllers/RatesController.cs:39]: It is more convenient to use OkOrBadRequest()

File: HappyTravel.Tsutsujigasaki.Api/Controllers/RatesController.cs (hunk 1 above)
Comment (DaniyarAskarov) [reply HappyTravel.Tsutsujigasaki.Api/Controllers/RatesController.cs:32]: This name already was in first method GetRate({sourceCurrency}/{targetCurrency}) and it takes more time to find and rename both methods.

File: HappyTravel.Tsutsujigasaki.Api/Controllers/RatesController.cs (hunk 2 above)
Comment (DaniyarAskarov) [reply HappyTravel.Tsutsujigasaki.Api/Controllers/RatesController.cs:33]: I thought about it. By the way first time when I started to do it I did it with nullable targetCurrencies. But it takes many difficulties to make it work well. In the end I desided that it's not worth it. There were many bugs.

**Review by AlexeyShok — APPROVED** [review]

**Pull request merged**